source/includes/connection-snippets/x509-snippets/c-connection.c

```diff
@@ -0,0 +1,37 @@
+// begin x509 connection
+#include <mongoc/mongoc.h>
+
+mongoc_client_t * client;
+mongoc_database_t * database;
+mongoc_collection_t * collection;
+bson_t * command, reply, * insert;
+bson_error_t error;
+char * str;
+bool retval;
+
+mongoc_init();
+
+const mongoc_ssl_opt_t * ssl_default = mongoc_ssl_opt_get_default();
+mongoc_ssl_opt_t ssl_opts = {
+  0
+};
+
+memcpy( & ssl_opts, ssl_default, sizeof ssl_opts);
+
+ssl_opts.pem_file = "/etc/certs/mongodb/client.pem";
+ssl_opts.ca_file = "/etc/certs/mongodb/ca.pem";
+ssl_opts.weak_cert_validation = false;
+ssl_opts.allow_invalid_hostname = false;
+
+char subject[] = "mongodb://<your_subject>"; // mongodb://CN%3DChris%2COU%3DTestClientCertificateOrgUnit%2CO%3DEducationClientCertificate%2CL%3DTestClientCertificateLocality%2CST%3DTestClientCertificateState%2CC%3DUS
+
+char uri[] = strcat(subject, "@localmongo1/?authMechanism=MONGODB-X509")
```

Comment:
Should we use `localmongo1` as the host? Perhaps there's a better standard for atlas urls?

Reply:
how about `mongodb+srv://<cluster-url>`? @ccho-mongodb

Reply:
Sure, whatever you're able to get driver teams to agree to. However, I think you meant 
```
mongodb+srv://<your x509 client subject>@<cluster-url>/?authMechanism=MONGODB-X509")
```

Reply:
Fixed. Thanks!